service/history/task/cross_cluster_source_task_executor.go

```diff
@@ -361,7 +404,7 @@ func (t *crossClusterSourceTaskExecutor) executeRecordChildWorkflowExecutionComp
 			types.CrossClusterTaskFailedCauseWorkflowAlreadyCompleted:
 			// Do nothing, these errors are expected if the target workflow is already closed
 		default:
-			return errUnexpectedErrorFromTarget
+			return errContinueExecution
```

Comment:
To fix this issue we also need to reset the task processing state before returning the error. Otherwise, it will be `reported` and the task is not available for polling.
`t.setTaskState(task, ctask.TaskStatePending, processingState(task.response.TaskState))`
I will land my pr after this one, so I can fix it.